wasm-node/src/node.rs

```diff
@@ -0,0 +1,187 @@
+use std::convert::Into;
+
+use anyhow::Context;
+use celestia_node::network::{canonical_network_bootnodes, network_genesis, network_id};
+use celestia_node::node::{Node, NodeConfig};
+use celestia_node::store::IndexedDbStore;
+use celestia_types::{hash::Hash, ExtendedHeader};
+use js_sys::Array;
+use libp2p::identity::Keypair;
+use libp2p::{identity, Multiaddr};
+use serde_wasm_bindgen::{from_value, to_value};
+use tracing::info;
+use wasm_bindgen::prelude::*;
+
+use crate::utils::Network;
+use crate::wrapper::libp2p::NetworkInfo;
+use crate::Result;
+
+#[wasm_bindgen(js_name = Node)]
+struct WasmNode {
+    node: Node<IndexedDbStore>,
+}
+
+#[wasm_bindgen(js_name = NodeConfig)]
+pub struct WasmNodeConfig {
+    pub network: Network,
+    #[wasm_bindgen(skip)]
+    pub genesis_hash: Option<Hash>,
+    #[wasm_bindgen(skip)]
+    pub p2p_local_keypair: Keypair,
+    #[wasm_bindgen(skip)]
+    pub p2p_bootnodes: Vec<Multiaddr>,
+}
+
+#[wasm_bindgen(js_class = Node)]
+impl WasmNode {
+    #[wasm_bindgen(constructor)]
+    pub async fn new(config: WasmNodeConfig) -> Self {
+        let network_id = network_id(config.network.into());
+        let store = IndexedDbStore::new(network_id).await.unwrap();
+        info!(
+            "Initialised store with head height: {:?}",
```

Comment:
this prints

> Initialised store with head height: Err(NotFound)

Reply:
Switched to printing head only,if it exists, same as native.